packages/nextjs/utils/scaffold-eth/networks.ts

```diff
@@ -166,14 +166,13 @@ export const getNetworkDetailsByChainId = (id: number) => Object.values(NETWORKS
  */
 export const getTargetNetwork = () => {
   const network = process.env.NEXT_PUBLIC_NETWORK;
-  const configuredChain = NETWORKS[network ?? "hardhat"];
 
-  if (!configuredChain) {
+  if (!network || !NETWORKS[network]) {
     // If error defaults to hardhat local network
     console.error("Network name misspelled or unsupported network used in process.env");
```

Comment:
Also should mention in the end of console.error that the "target network is set to hardhat" OR "using default network as hardhat"

Reply:
And probably change console.error to `Network name is not set, misspelled or unsupported network used in process.env`